src/goto-instrument/contracts/contracts.cpp

```diff
@@ -1260,3 +1258,63 @@ bool code_contractst::enforce_contracts(const std::set<std::string> &to_enforce)
   }
   return fail;
 }
+
+void code_contractst::find_static_locals(
```

Comment:
@tautschnig is there anything already available in CBMC that could helps us to get all static local symbols of a given function?

Reply:
I was thinking of something simpler like:

```c++
for(const auto &foo : call_tree(foo))
  for(const auto &sym_pair : symbol_table)
    if(sym.is_static_lifetime && sym.location.get_function() == foo)
      local_statics.add(sym)
```

to find all local `static`s indirectly referred to by `foo`. Do we have a CBMC library function that returns a call tree below a given function `foo`?

Reply:
now using the call_graph of the function under analysis to perform the detection